implementation.js

```diff
@@ -162,13 +164,11 @@ module.exports = function from(arrayLike) {
 		len = arrayFromIterable.length;
 	} else if (isString(items)) {
 		return strMatch.call(items, /[\uD800-\uDBFF][\uDC00-\uDFFF]?|[^\uD800-\uDFFF]|./g) || [];
-	} else if (forOfOnly) {
-		try {
-			// Safari 8's native Map or Set can't be iterated except with for..of
-			return Function('items', 'arrayLike', 'var arr = []; for (var entry of arrayLike) { arr.push(entry); } return arr')(); // eslint-disable-line no-new-func
-		} catch (e) {
-			/* for..of seems to not be supported, fallback to basic iteration */
-		}
+	} else if (forOf) {
```

Comment:
since you have `forOf` set to a default identity functiom, this will always be truthy - shouldn't it just be `var forOf;` up top?